sympy/polys/polytools.py

```diff
@@ -4167,6 +4167,9 @@ def _strict_eq(f, g):
 class PurePoly(Poly):
     """Class for representing pure polynomials. """
 
+    def __init__(self, rep, *gens, **args):
```

Comment:
Since `PurePoly` is a `Basic` subclass I don't think it should have an `__init__` method.

Reply:
I have understood that a class can have both a `__new__` and an `__init__` method. A subclass could add something in the `__init__` method.

Reply:
IMO, objects can be both created and initialised in `__new__`, though for clarity we can use `__init__` in subclasses if there isn't going to be any change in `__new__` method. I am not sure which object will be returned by  `__new__` in this case. May it should be checked.